### ISSUE TITLE
azure.azcollection.azure_rm_snapshot snapshot creation fails

### ISSUE THREAD

**Issue by Alffernandez** [ansible/ansible_collections_azure#1478]
##### SUMMARY

azure_rm_snapshot fails with the following error:
```
module_stderr: |- Traceback (most recent call last): File "/tmp/.ansible_/tmp/ansible-tmp-1709387186.9568183-626-95902907116320/AnsiballZ_azure_rm_snapshot.py", line 107, in <module> _ansiballz_main() File "/tmp/.ansible_/tmp/ansible-tmp-1709387186.9568183-626-95902907116320/AnsiballZ_azure_rm_snapshot.py", line 99, in _ansiballz_main invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS) File "/tmp/.ansible_/tmp/ansible-tmp-1709387186.9568183-626-95902907116320/AnsiballZ_azure_rm_snapshot.py", line 47, in invoke_module runpy.run_module(mod_name='ansible_collections.azure.azcollection.plugins.modules.azure_rm_snapshot', init_globals=dict(_module_fqn='ansible_collections.azure.azcollection.plugins.modules.azure_rm_snapshot', _modlib_path=modlib_path), File "<frozen runpy>", line 226, in run_module File "<frozen runpy>", line 98, in _run_module_code File "<frozen runpy>", line 88, in _run_code File "/tmp/ansible_azure_rm_snapshot_payload_16oxyv7q/ansible_azure_rm_snapshot_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_snapshot.py", line 398, in <module> File "/tmp/ansible_azure_rm_snapshot_payload_16oxyv7q/ansible_azure_rm_snapshot_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_snapshot.py", line 394, in main File "/tmp/ansible_azure_rm_snapshot_payload_16oxyv7q/ansible_azure_rm_snapshot_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_snapshot.py", line 217, in __init__ File "/tmp/ansible_azure_rm_snapshot_payload_16oxyv7q/ansible_azure_rm_snapshot_payload.zip/ansible_collections/azure/azcollection/plugins/module_utils/azure_rm_common.py", line 469, in __init__ File "/tmp/ansible_azure_rm_snapshot_payload_16oxyv7q/ansible_azure_rm_snapshot_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_snapshot.py", line 242, in exec_module File "/tmp/ansible_azure_rm_snapshot_payload_16oxyv7q/ansible_azure_rm_snapshot_payload.zip/ansible_collections/azure/azcollection/plugins/module_utils/azure_rm_common.py", line 948, in get_mgmt_svc_client AttributeError: 'GenericRestClient' object has no attribute '_config'. Did you mean: 'config'?
```
##### ISSUE TYPE
- Bug Report

##### ANSIBLE VERSION

```paste below
ansible-playbook [core 2.15.9] 
```

##### COLLECTION VERSION

```paste below
azure.azcollection 2.2.0
```

##### CONFIGURATION

```paste below

```

##### OS / ENVIRONMENT

```
NAME="Red Hat Enterprise Linux"
VERSION="8.8 (Ootpa)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="8.8"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Red Hat Enterprise Linux 8.8 (Ootpa)"
```
##### STEPS TO REPRODUCE

```yaml
1. Create an execution environment with azure.azcollection 2.2.0 and all its dependencies. Or install azure.azcollection 2.2.0 and all its dependencies directly.

2. Run the following playbook:

  name: Azure snapshot
  hosts: all
  tasks:
   - name: Create a snapshot by copying an existing managed disk.
     azure_rm_snapshot:
       resource_group: "{{ vm_details.resource_group|lower }}"
       subscription_id: "{{ vm_details.subscription_id }}"
       tenant: "{{ azuzones[vm_details.subscription_id].tenant}}"
       secret: "{{ azuzones[vm_details.subscription_id].secret}}"
       client_id:  "{{ azuzones[vm_details.subscription_id].client_id}}"
       name: mySnapshot
       cert_validation_mode: ignore
       sku:
         name: Standard_ZRS
       creation_data:
         create_option: Copy
         source_id: "{{ item }}"
       append_tags: True
       tags:
         test: "test"

3. Get the following error:

AttributeError: 'GenericRestClient' object has no attribute '_config'. Did you mean: 'config'?
```
This error is not happening with much older azure.azcollection versions.

Also, a similar error was reported in the following Github ticket:

https://github.com/ansible-collections/azure/issues/827

After adding  the certificates and removed the option to ignore certs and the error has been fixed so, the error looks appearing when the certificates are ignored.

##### EXPECTED RESULTS

```
Able to perform snapshots with the latest azure.azcollection version.
```
##### ACTUAL RESULTS

```paste below
Unable to perform snapshots with the latest azure.azcollection version.
```

**Comment by Fred-sun** [comment]
@Alffernandez I'm very sorry that after upgrading to the SDK of track2, I haven't found a way to replace clinet._config. I'm still looking for a way to solve this problem. Please do not add parameter 'cert_validation_mode' to playbook until the problem is solved. Thank you!

**Comment by Fred-sun** [comment]
@Alffernandez Fixed by #1554, Please take a try! Thank you very much!
